src/App.jsx

```diff
@@ -13,34 +14,87 @@ export const goodsFromServer = [
   'Jam',
   'Garlic',
 ];
+const SORT_BY_ALPHABET = 'alphabet';
+const SORT_BY_LENGTH = 'length';
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const getPreparedGoods = (goods, sorter) => {
+  const { sortField, isReverse } = sorter;
+  const prepGoods = [...goods];
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  if (sortField === SORT_BY_ALPHABET) {
+    prepGoods.sort((a, b) => a.localeCompare(b));
+  }
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  if (sortField === SORT_BY_LENGTH) {
+    prepGoods.sort((a, b) => a.length - b.length);
+  }
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  if (isReverse) {
+    prepGoods.reverse();
+  }
+
+  return prepGoods;
+};
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [isReverse, setIsReverse] = useState(false);
+  const visibleGoods = getPreparedGoods(goodsFromServer, {
+    sortField,
+    isReverse,
+  });
+
+  const handleReset = () => {
+    setIsReverse(false);
+    setSortField('');
+  };
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={`button is-info ${sortField === SORT_BY_ALPHABET ? '' : 'is-light'}`}
+          onClick={() => setSortField(SORT_BY_ALPHABET)}
+        >
+          Sort alphabetically
+        </button>
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+        <button
+          type="button"
+          className={`button is-success is-light" ${sortField === SORT_BY_LENGTH ? '' : 'is-light'}`}
```

Comment:
There is a minor issue with the className assignment for the 'Sort by length' button. The string `is-light` is repeated, which may cause unintended styling issues. Consider removing the extra `is-light` from the className.